static/js/src/prepare-form-inputs.js

```diff
@@ -0,0 +1,139 @@
+export function setupIntlTelInput(countryCode, phoneInput) {
+  console.log("Setting up intlTelInput with", phoneInput);
+  phoneInput.removeAttribute("name"); // Ensure only the hidden input is submitted.
+
+  intlTelInput(phoneInput, {
+    utilsScript: "/static/js/dist/utils.js",
+    separateDialCode: true,
+    hiddenInput: phoneInput.name,
+    initialCountry: countryCode,
+  });
+
+  addInputValidation(phoneInput);
+}
+
+/**
+ * Adds validation logic and styling to the phone input field.
+ *
+ * @param {HTMLElement} phoneInput - The input element for the phone number.
+ */
+function addInputValidation(phoneInput) {
+  const mobileInput = document.querySelector(".iti");
+  mobileInput.parentNode.classList.add("p-form-validation");
+  phoneInput.classList.add("p-form-validation__input");
+  phoneInput.setAttribute("aria-describedby", "invalid-number-message");
+
+  const errorElement = createErrorMessage();
+  phoneInput.addEventListener("blur", () =>
+    validateInput(phoneInput, errorElement)
+  );
+  phoneInput.addEventListener("change", () => resetErrorState(errorElement));
+  phoneInput.addEventListener("keyup", () => resetErrorState(errorElement));
+}
+
+/**
+ * Creates and returns an error message.
+ *
+ * @returns {HTMLElement} errorElement - The error message element.
+ *
+ */
+function createErrorMessage() {
+  const errorElement = document.createElement("div");
+  errorElement.id = "invalid-number-message";
+  errorElement.className = "p-form-validation__message";
+  errorElement.style.marginTop = "1rem";
+  errorElement.textContent = "Please enter a valid number.";
+  errorElement.setAttribute("role", "alert");
+  return errorElement;
+}
+
+/**
+ * Validates the phone number and shows or hides the error message accordingly.
+ *
+ * @param {HTMLElement} errorElement - The error message element.
+ */
+function validateInput(phoneInput, errorElement) {
+  resetErrorState(errorElement, phoneInput);
+  const isValid = isValidNumber(phoneInput.value.trim());
+  if (!isValid) {
+    phoneInput.parentNode.parentNode.classList.add("is-error");
+    phoneInput.parentNode.after(errorElement);
+  }
+}
+
+/**
+ * Checks the phone number is valid.
+ *
+ * @param {string} number - The phone number to validate.
+ * @returns {boolean} - True if the number is valid, otherwise false.
+ */
+function isValidNumber(number) {
+  const pattern = /^(?=[^a-zA-Z]*$)[0-9\s.\-()/,]{4,25}$/;
+  return pattern.test(number);
+}
+
+/**
+ * Resets the error states.
+ *
+ * @param {HTMLElement} errorElement - The error message element to remove.
+ */
+function resetErrorState(errorElement, phoneInput) {
+  phoneInput.parentNode.parentNode.classList.remove("is-error");
+  if (errorElement.parentNode) {
+    errorElement.remove();
+  }
+}
+
+const phoneNumberInput = document.querySelector("input#phone");
+const countryInput = document.querySelector("select#country");
+
+if (phoneNumberInput || countryInput) {
+  prepareInputFields(phoneNumberInput, countryInput);
```

Comment:
can you please add a comment in this block explaining what it is doing? I got a bit confused when trying to understand it. In places where you import `prepare-form-inputs.js` and call `prepareInputFields` you already pass selectors for phoneNumberInput and countryInput. Then why do you need to call `prepareInputFields` here again?

Reply:
I have added a comment, but essentially it exists so this can be used in both static and modal forms